Chocolatey-AU.build.ps1

```diff
@@ -0,0 +1,375 @@
+        }
+        else {
+            'http://timestamp.digicert.com'
+        }
+    ),
+
+    [string]
+    $CertificatePath = $env:CHOCOLATEY_OFFICIAL_CERT,
+
+    [string]
+    $CertificatePassword = $env:CHOCOLATEY_OFFICIAL_CERT_PASSWORD,
+
+
+    [string]
+    $CertificateAlgorithm = $(
+        if ($env:CERT_ALGORITHM) {
+            $env:CERT_ALGORITHM
+        }
+        else {
+            'Sha256'
+        }
+    ),
+
+    [string]
+    $CertificateSubjectName = "Chocolatey Software, Inc.",
+
+
+    [string]
+    $NugetApiKey = $env:POWERSHELLPUSH_API_KEY,
+
+
+    [string]
+    $PublishUrl = $env:POWERSHELLPUSH_SOURCE,
+
+
+    [string]
+    $ModuleName = 'Chocolatey-AU',
+
+
+    [switch]
+    $ThrowOnPSSAViolation
+)
+
+$ErrorActionPreference = 'Stop'
+
+$script:SourceFolder = "$PSScriptRoot/src"
+$script:ReleaseBuild = -not [string]::IsNullOrEmpty((git tag --points-at HEAD 2> $null) -replace '^v')
+$script:BuildVersion = $null
+$script:IsPrerelease = $false
+$script:ModuleOutputDir = "$OutputDirectory/$ModuleName"
+
+
+# Fix for Register-PSRepository not working with https from StackOverflow:
+# https://stackoverflow.com/questions/35296482/invalid-web-uri-error-on-register-psrepository/35296483#35296483
+function Register-PSRepositoryFix {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $Name,
+
+        [Parameter(Mandatory = $true)]
+        [Uri]
+        $SourceLocation,
+
+        [ValidateSet('Trusted', 'Untrusted')]
+        $InstallationPolicy = 'Trusted'
+    )
+
+    $ErrorActionPreference = 'Stop'
+
+    try {
+        Write-Verbose 'Trying to register via Register-PSRepository'
+        Register-PSRepository -Name $Name -SourceLocation $SourceLocation -InstallationPolicy $InstallationPolicy -ErrorAction Stop
+        Write-Verbose 'Registered via Register-PSRepository'
+    }
+    catch {
+        Write-Verbose 'Register-PSRepository failed, registering via workaround'
+
+        # Adding PSRepository directly to file
+        Register-PSRepository -Name $Name -SourceLocation $env:TEMP -InstallationPolicy $InstallationPolicy -ErrorAction Stop
+        $PSRepositoriesXmlPath = "$env:LOCALAPPDATA\Microsoft\Windows\PowerShell\PowerShellGet\PSRepositories.xml"
+        $repos = Import-Clixml -Path $PSRepositoriesXmlPath
+        $repos[$Name].SourceLocation = $SourceLocation.AbsoluteUri
+        $repos[$Name].PublishLocation = [uri]::new($SourceLocation, 'package').AbsoluteUri
+        $repos[$Name].ScriptSourceLocation = ''
+        $repos[$Name].ScriptPublishLocation = ''
+        $repos | Export-Clixml -Path $PSRepositoriesXmlPath
+
+        # Reloading PSRepository list
+        Set-PSRepository -Name PSGallery -InstallationPolicy Trusted
+        Write-Verbose 'Registered via workaround'
+    }
+}
+
+# Synopsis: ensure GitVersion is installed
+task InstallGitVersion {
+    if (-not (Get-Command gitversion -ErrorAction Ignore)) {
```

Comment:
GitVersion can also be installed through the `dotnet` utility, do we also want to be able to handle that? (I think the name in that case is `dotnet-version`, but I could be wrong).

Reply:
I like the idea of checking and using it if it's installed, but I'm not sure on installing the dotnet tool here. It feels like the logic for it will get too complicated. But the logic to use it if it's already there is fairly straightforward.

Reply:
The idea was not to install it through dotnet, only check the presence and use it if it is installed. But it is not something that would stop this PR, but could be an enhancement for the future.